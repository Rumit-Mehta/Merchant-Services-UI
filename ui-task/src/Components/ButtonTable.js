import React from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";
import CancelButton from "./CancelButton/CancelButton";
import SubmitButton from "./AssignServicesButton/AssignServicesButton";

export default function buttonTable() {
  return (
    <div className="buttonDiv">
      <Table id="button.table">
        <TableBody>
          <TableRow>
            <TableCell align="left">
              <CancelButton />
            </TableCell>
            <TableCell align="right">
              <SubmitButton />
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
}
