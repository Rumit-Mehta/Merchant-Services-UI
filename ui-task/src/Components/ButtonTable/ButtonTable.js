import React from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";
import CancelButton from "../CancelButton/CancelButton";
import SubmitButton from "../AssignServicesButton/AssignServicesButton";

// Default function that returns a table with 1 row and 2 columns, 1 for each of the buttons
// made to be able to easily layout the buttons (AssignServicesButton && CancelButton)
export default function buttonTable() {
  return (
    <div className="buttonDiv">
      <Table id="button.table">
        <TableBody>
          <TableRow>
            <TableCell align="left">
              <CancelButton />
            </TableCell>
            <TableCell align="right">
              <SubmitButton />
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
}
