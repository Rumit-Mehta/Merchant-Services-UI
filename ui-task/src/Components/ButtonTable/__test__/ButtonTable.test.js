import React from "react";
import ReactDOM from "react-dom";
import ButtonTable from "../ButtonTable";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { TableRow, TableCell } from "@material-ui/core";

afterEach(cleanup);
test("renders  without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ButtonTable />, div);
});

test("table has correct rows and columns", () => {
  const wrapper = shallow(<ButtonTable />);
  const rows = wrapper.find(TableRow);
  const columns = wrapper.find(TableCell);
  expect(rows).toHaveLength(1);
  expect(columns).toHaveLength(2);
});

test("matches snapshot", () => {
  const tree = renderer.create(<ButtonTable />).toJSON();
  expect(tree).toMatchSnapshot();
});
