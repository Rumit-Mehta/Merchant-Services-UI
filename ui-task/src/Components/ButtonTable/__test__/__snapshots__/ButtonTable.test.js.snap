// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="buttonDiv"
>
  <table
    className="MuiTable-root"
    id="button.table"
    role={null}
  >
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          <div
            data-testid="cancelButtonDiv"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Cancel
              </span>
            </button>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <div
            className="serviceDiv"
            data-testid="assignServiceDiv"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text assignServiceButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "#1176b6",
                  "border": 0,
                  "borderRadius": 100,
                  "color": "white",
                  "height": 48,
                  "padding": "0 30px",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Assign Services
              </span>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
